modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java

```diff
@@ -1107,6 +1112,45 @@ public static boolean isCommandNoOp(Prepared cmd) {
         return cmd instanceof NoOperation;
     }
 
+    /**
+     * Creates table digest as MD5 hash from sorted list of column names and types.
+     *
+     * @param tbl Create table command
+     * @return Digest from sorted list of column names and types.
+     */
+    private static String createFieldsDigest(GridSqlCreateTable tbl) {
+        try {
+            List<String> fieldDigests = new ArrayList<>(tbl.columns().size());
+
+            for (Map.Entry<String, GridSqlColumn> e : tbl.columns().entrySet()) {
+                String colName = e.getKey();
+                String colType = getTypeClassName(e.getValue());
+
+                String fd = "[" + colName + ":" + colType + "]";
+
+                fieldDigests.add(fd.toUpperCase());
+            }
+
+            Collections.sort(fieldDigests);
+
+            String fieldsDigest = String.join(", ", fieldDigests);
+
+            MessageDigest md = MessageDigest.getInstance("MD5");
+
+            md.update(fieldsDigest.getBytes());
+
+            StringBuilder sb = new StringBuilder();
+
+            for (byte md5Byte : md.digest())
```

Comment:
Do you mean Integer.toHexString(md5Byte) here?
You can use Hex.encodeHex(byte[]) from apache-commons which look less tricky and generate much less garbage.

Reply:
also _IgniteUtils_ already has `public static String byteArray2HexString(byte[] arr)`. It may suits.

Reply:
Unfortunately, we do not have dependency on apache-commons library in Ignite's core module and therefore I can not use `Hex.encodeHex(byte[])` here. Moreover, this is not a hot path, so garbage generation impact can be neglected here.

Reply:
There is commons-codec dependency in ignite-indexing.
CommandProcessor is a part of indexing module, not core.